### PR TITLE
WIP : allocations in VCAB3

### PR DESCRIPTION
**Pull request by sipah00**
#346

### CODE REVIEW

File: src/adams_utils.jl [hunk 1]
```diff
@@ -3,8 +3,15 @@ function ϕ_and_ϕstar!(cache, dy, next_point, last_idx)
   for i = 0:(k)-1
     if i == 0
       β[(i)+1] = 1
-      ϕ_n[(i)+1] = copy(dy)
-      ϕstar_n[(i)+1] = copy(dy)
+      if typeof(dy) <: Array
+        # ϕ_n[(i)+1] = copy(dy)
+        ϕ_n[(i)+1] .= dy
+        # ϕstar_n[(i)+1] .= dy
```
Comment (sipah00) [inline src/adams_utils.jl]: @ChrisRackauckas Why broadcasting is not working (convergence tests are failing) here. I could not figure out what is the problem.

Comment (americast) [reply src/adams_utils.jl]: :thinking:

Comment (ChrisRackauckas) [reply src/adams_utils.jl]: so it passes when it's a copy?

Comment (ChrisRackauckas) [reply src/adams_utils.jl]: Are all of the `ϕstar_n` vectors initialized in a way that they alias?

Comment (sipah00) [reply src/adams_utils.jl]: yeah

File: src/caches/adams_bashforth_moulton_caches.jl [hunk 2]
```diff
@@ -349,12 +349,17 @@ function alg_cache(alg::VCAB3,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUni
   k2 = zeros(rate_prototype)
   k3 = zeros(rate_prototype)
   k4 = zeros(rate_prototype)
-  ϕstar_nm1 = Array{typeof(rate_prototype)}(1,3)
   grid_points = zeros(typeof(t),3)
   c = zeros(typeof(t),3,3)
   g = zeros(typeof(t),3)
   ϕ_n = Array{typeof(rate_prototype)}(1,3)
+  ϕstar_nm1 = Array{typeof(rate_prototype)}(1,3)
   ϕstar_n = Array{typeof(rate_prototype)}(1,3)
```
Comment (sipah00) [inline src/caches/adams_bashforth_moulton_caches.jl]: @ChrisRackauckas initialization

File: src/caches/adams_bashforth_moulton_caches.jl [hunk 3]
```diff
@@ -349,12 +349,17 @@ function alg_cache(alg::VCAB3,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUni
...
+  for i in 1:3
+    ϕ_n[i] = zeros(rate_prototype)
+    ϕstar_nm1[i] = zeros(rate_prototype)
+    ϕstar_n[i] = zeros(rate_prototype)
```
Comment (ChrisRackauckas) [inline src/caches/adams_bashforth_moulton_caches.jl]: doesn't look like these should alias. Try printing out the `ϕstar_n` values and see if every vector is the same.

**Pull request merged**